dev/src/container.rs

```diff
@@ -206,3 +210,32 @@ pub async fn pull_image(docker: &Docker, name: &str) -> Result<()> {
 
     Ok(())
 }
+
+fn random_lowercase_string(len: i32) -> String {
+    let mut rng = rand::thread_rng();
+    let alphabet = Uniform::from('a'..='z');
+    (0..len).map(|_| rng.sample(alphabet) as char).collect()
+}
+
+pub async fn build_image<P: AsRef<Path>>(path: P) -> Result<String> {
+    // the docker api requires tarballs, which is just an unnecessary headache, so just using the binary here.
```

Comment:
yeah, that's a whole rabbit hole with implementing `.dockerignore` handling etc., calling the binary is 100% better.